bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/util/mapper/ActionInputsToConfigDescriptionParameters.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.automation.util.mapper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.measure.Quantity;
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.automation.type.Input;
+import org.openhab.core.config.core.ConfigDescriptionParameter;
+import org.openhab.core.config.core.ConfigDescriptionParameterBuilder;
+import org.openhab.core.i18n.UnitProvider;
+import org.openhab.core.types.util.UnitUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a utility class to convert action {@link Input}s to {@link ConfigDescriptionParameter}s.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+@NonNullByDefault
+public class ActionInputsToConfigDescriptionParameters {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ActionInputsToConfigDescriptionParameters.class);
+
+    public static final Pattern QUANTITY_TYPE_PATTERN = Pattern
+            .compile("([a-z0-9]+\\.)*QuantityType<([a-z0-9]+\\.)*(?<dimension>[A-Z][a-zA-Z0-9]*)>");
+
+    /**
+     * Maps a list of {@link Input}s to a list of {@link ConfigDescriptionParameter}s.
+     *
+     * @param inputs the list of inputs to map to config description parameters
+     * @param unitProvider the unit provider to be used to get the default unit for a dimension, can be null
+     * @return the list of config description parameters or null if an input parameter has an unsupported type
+     */
+    public static @Nullable List<ConfigDescriptionParameter> map(List<Input> inputs,
+            @Nullable UnitProvider unitProvider) {
```

Comment:
What sitiation results in the `unitProvider` being `null`? In general: same comment as for `AnnotationActionModuleTypeHelper`: can't this be a component with the `UnitProvider` injected in the constructor?

Reply:
To be checked if all consumers are components.
Do you think I should merge the two classes in one unique component? Or is it fine to have two new components?

Reply:
Yes it all consumers are components.

Reply:
Creating a component for `ActionInputsToConfigDescriptionParameters` seems doable.

Reply:
Yes, everything that is related to action input should/could be in one component. The `AnnotationActionModuleTypeHelper` could stay as a separate component. I would also suggest to name the new component with `Helper` at the end (to keep naming consistent).

Reply:
One helper component created